eng/common/scripts/Delete-RemoteBranches.ps1

```diff
@@ -1,46 +1,69 @@
   # E.g Pull request opened from azure-sdk/azure-sdk prBranch --> Azure/azure-sdk baseBranch
-  $ForkRepoOwner,
-  [Parameter(Mandatory = $true)]
-  $RepoName,
+  $ForkRepoName,
   [Parameter(Mandatory = $true)]
   $BranchPrefix,
-  [Parameter(Mandatory = $true)]
-  $AuthToken
+  # Date format: Tuesday, April 12, 2022 1:36:02 PM
+  $LastCommitOlderThan
 )
 
 . (Join-Path $PSScriptRoot common.ps1)
 
-LogDebug "Operating on Repo [ $RepoName ]"
-try{
-  $branches = (Get-GitHubSourceReferences -RepoOwner $RepoOwner -RepoName $RepoName -Ref "heads/$BranchPrefix" -AuthToken $AuthToken).ref 
+LogDebug "Operating on Repo [ $Repo ]"
+if (($Repo -split "/").Count -lt 2) {
+  LogError "The $Repo does not have right format. Please use RepoOwner/RepoName. E.g. Azure/azure-sdk-for-java. Your input: $Repo"
+  exit 1
 }
-catch {
-  LogError "Get-GitHubSourceReferences failed with exception:`n$_"
+$RepoOwner = ($Repo -split "/")[0]
+$RepoName = ($Repo -split "/")[1]
+$GithubAPIBaseURI = "repos/$RepoOwner/$RepoName"
+
+function HandleGHAPICall ($api, $method="GET") {
+  $err = $( $responseJson = gh api -X $method $api ) 2>&1
+
+  if ($LASTEXITCODE) {
+      Write-Host "API call `gh api -X $method $api` returned error: $err"
+      return $null
+  }
+  return $responseJson
+}
+
+# Get all branches with the branch prefixes. 
+$GetBranchesGithubAPI = "$GithubAPIBaseURI/git/matching-refs/heads/$BranchPrefix"
+$responseJson = HandleGHAPICall -api $GetBranchesGithubAPI
+if ($null -eq $responseJson) {
+  Write-Error "Error calling the get branch API. Exiting..."
   exit 1
 }
+# Throw the error out if the response is not in json format and failed the script.
+$responses = ConvertFrom-Json $responseJson
 
-foreach ($branch in $branches)
+foreach ($res in $responses)
 {
-  try {
-    $branchName = $branch.Replace("refs/heads/","")
-    $head = "${RepoOwner}/${RepoName}:${branchName}"
-    LogDebug "Operating on branch [ $branchName ]"
-    $pullRequests = Get-GitHubPullRequests -RepoOwner $RepoOwner -RepoName $RepoName -State "all" -Head $head -AuthToken $AuthToken
-
-    if ($ForkRepoOwner)
-    {
-      $pullRequests += Get-GitHubPullRequests -RepoOwner $ForkRepoOwner -RepoName $RepoName -State "all" -Head $head -AuthToken $AuthToken
-    }
+  $branch = $res.ref
+  if (!$branch) {
+    LogDebug "No branch returned from the get branch respones."
+    exit 0
   }
-  catch
-  {
-    LogError "Get-GitHubPullRequests failed with exception:`n$_"
+  $branchName = $branch.Replace("refs/heads/","")
+  $head = "${RepoOwner}/${RepoName}:${branchName}"
+  LogDebug "Operating on branch [ $branchName ]"
+  # Get PR associate with the branch.
+  $GetPullRequestFromBranchAPI = "$GithubAPIBaseURI/pulls?state=all&head=$head"
+  $responseJson = HandleGHAPICall -api $GetPullRequestFromBranchAPI
+  if ($null -eq $responseJson) {
+    LogError "Error calling the get pullrequest API. Exiting..."
     exit 1
   }
+  # The piece of logic did not in use today. 
+  # if ($ForkRepoName)
```

Comment:
Did not find the usage of the ForkRepoName. @weshaggard @chidozieononiwu Could you suggest how it gets used and do we need to leave the code?

Reply:
We used to have the branches in the fork as well so I think it was about that. I think we can remove this now and if we need to clean up forks we can add them to the list.